dojo/tools/checkmarx_osa/parser.py

```diff
@@ -0,0 +1,95 @@
+import json
+import logging
+from dojo.models import Finding
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+
+class CheckmarxOsaParser(object):
+
+    def get_scan_types(self):
+        return ["Checkmarx OSA"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Checkmarx Open Source Analysis for dependencies (json). Generate with `jq -s . CxOSAVulnerabilities.json CxOSALibraries.json`"
+
+    def get_findings(self, filehandle, test):
+        tree = json.load(filehandle)
+        if len(tree) != 2:
+            logger.error("Bad format. Expected a list of two elements: CxOSALibraries.json and CxOSAVulnerabilities.json. Found %i elements", len(tree))
+            raise Exception("Invalid format: bad structure")
+        libraries_dict = self.get_libraries(tree)
+        vulnerabilities = self.get_vunlerabilities(tree)
+        items = []
+        for item in vulnerabilities:
+            library = libraries_dict[item['libraryId']]
+            # Possible status as per checkmarx 9.2: TO_VERIFY, NOT_EXPLOITABLE, CONFIRMED, URGENT, PROPOSED_NOT_EXPLOITABLE
+            status = item['state']['name']
+            finding_item = Finding(
+                title='{0} {1} | {2}'.format(library['name'], library['version'], item['cveName']),
+                severity=item['severity']['name'],
+                description=item['description'],
+                unique_id_from_tool=item['id'],
```

Comment:
Could we add 'libraryId' as vuln_type_id_from_tool?

Reply:
you're suggesting added a field to track the technical id from checkmarx OSA that identifies that specific library ? to what purpose? the component_name and component_version should be enough for us I believe

Reply:
Maybe I didn't understand well what 'libraryId' is. Forget my comment.